src/lib.rs

```diff
@@ -206,6 +206,12 @@ impl StarAligner {
         rec
     }
 
+    fn empty_records(name: &[u8], read: &[u8], qual: &[u8], n: usize) -> Vec<bam::Record> {
```

Comment:
Looking at this, do we have a few other issues?
I think the empty records should have:
- 0x4 / 0x8 flags set to declare whether they are R1 or R1.
- Exactly one of the records must be primary (flags & 0x900 == 0), and should probably form a pair with the primary record that was returned by STAR.
I think we might confuse bamtofastq without those properties.

Reply:
Good call.

I have fixed this up below (when we know whether first or last segment in the template).